src/plugins/intel_gna/src/backend/gna_limitations.cpp

```diff
@@ -677,7 +677,8 @@ constexpr uint32_t Limitations::kBytesPerCropElement;
 constexpr uint32_t Limitations::kBytesPerConcatElement;
 constexpr uint32_t Limitations::kMemoryPageSize;
 
-thread_local std::shared_ptr<Limitations> Limitations::k_instance{nullptr};
+std::unordered_map<std::thread::id, std::shared_ptr<Limitations>> Limitations::kInstances;
+std::mutex Limitations::kInstancesMtx;
```

Comment:
can we use ov::ThreadLocal https://github.com/openvinotoolkit/openvino/blob/master/src/inference/dev_api/openvino/runtime/threading/thread_local.hpp#L41 ?

I suppose it does not have issue with AppVerif (could you please double check it?)

Reply:
Thanks for suggestion. I have created a separate pull request with an implementation that utilizes InferenceEngine::ThreadLocal instead of the local_thread at https://github.com/openvinotoolkit/openvino/pull/21284. Once it passes the test with AppVerif, I will close this pull request.